src/NuGet.Services.AzureSearch/Analysis/TruncateCustomTokenFilter.cs

```diff
@@ -0,0 +1,20 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using Microsoft.Azure.Search.Models;
+
+namespace NuGet.Services.AzureSearch
+{
+    /// <summary>
+    /// Truncates tokens to 300 characters or less. This is necessary as Azure Search's
+    /// <see cref="TruncateTokenFilter"/> defaults to 10 characters.
+    /// </summary>
+    public static class TruncateCustomTokenFilter
+    {
+        public const string Name = "nuget_truncate_filter";
+
+        public static TruncateTokenFilter Instance = new TruncateTokenFilter(
+            Name,
+            length: 300);
```

Comment:
Any reason you chose 300?

Reply:
Two reasons:

1. It's the max allowed length for a token 
2. It's the documented default value of Azure Search's `TruncateTokenFilter` length (see [this](https://docs.microsoft.com/en-us/azure/search/index-add-custom-analyzers))... but their documentation is wrong 😢

Reply:
should be reported. :(

Reply:
I've sent the Azure Search team an email about it